clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -2794,6 +2794,33 @@ check_read_write_records_match_operands()
                          "Fail to catch missing write records"))
             return false;
     }
+    // Correct: skip read and write records check when the previous instruction was
+    // preempted.
+    {
+        instr_t *nop = XINST_CREATE_nop(GLOBAL_DCONTEXT);
+        instr_t *store = XINST_CREATE_store(
+            GLOBAL_DCONTEXT, OPND_CREATE_MEMPTR(REG2, /*disp=*/0), opnd_create_reg(REG1));
+        instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+        instrlist_append(ilist, store);
+        instrlist_append(ilist, nop);
+
+        std::vector<memref_with_IR_t> memref_instr_vec = {
+            { gen_marker(TID_A, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+              nullptr },
+            { gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64), nullptr },
+            { gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096), nullptr },
+            { gen_instr(TID_A), store },
+            { gen_marker(TID_A, TRACE_MARKER_TYPE_KERNEL_EVENT, 2), nop },
```

Comment:
I think raw2trace needs to remove the store instruction in this case.  For an asynchronous signal we should not have a partially complete instruction with an instruction record but no address record (for a synchronous signal it's less clear what should be done: there is an issue on it somewhere): that will just be confusing and violate our invariants without benefit as it's the same as the signal arriving completely before the store.  The store will be executed as the resumption point of the signal.

raw2trace has logic trying to place the signal properly: I think there must be a bug in that logic or it fails to consider this case?